components/brave_referrals/browser/BUILD.gn

```diff
@@ -18,6 +16,8 @@ source_set("browser") {
       "//brave/common",
       "//brave/components/brave_referrals/common",
       "//brave/components/brave_stats/browser",
+      "//brave/components/private_channel",
+      "//brave/components/private_channel/rust/ffi",
```

Comment:
This is not correct and will cause problems on macos where this is a shared library. rust/ffi should only be a dep of `brave_rust` and `"//brave/build/rust",` should be a public dep of private_channel

Reply:
- [ ] Remove ` "//brave/components/private_channel/rust/ffi",` from dependency from `brave_referrals`

Reply:
- [ ] `"//brave/components/private_channel/rust/ffi"` as dep of `brave_rust` and set `//brave/build/rust",` as public dep of private_channel